messaging/src/main/java/io/lantern/messaging/CryptoWorker.kt

```diff
@@ -576,8 +576,9 @@ internal class CryptoWorker(
             val paddedPlainText = Padding.padMessage(plainText)
             val to =
                 SignalProtocolAddress(recipientIdentityKey, deviceId)
-            val unidentifiedSenderMessage: ByteArray =
+            val unidentifiedSenderMessage: ByteArray = db.mutate {
```

Comment:
cipher.encrypt involves multiple calls to our ProtocolStore. It seems safest to perform all of them in a single transaction to leave the ProtocolStore in a consistent state in case something goes wrong.